engine/src/main/java/org/terasology/rendering/logic/FloatingTextRenderer.java

```diff
@@ -100,24 +93,23 @@ private void render(Iterable<EntityRef> floatingTextEntities) {
 
             FloatingTextComponent floatingText = entity.getComponent(FloatingTextComponent.class);
 
-            List<String> text = Arrays.asList(floatingText.text.split("\n"));
+            String[] linesOfText = floatingText.text.split("\n");
             Color baseColor = floatingText.textColor;
             Color shadowColor = floatingText.textShadowColor;
             boolean underline = false;
 
-            String longest = "";
-            for (String s : text) {
-                if (s.length() > longest.length())
-                    longest = s;
+            int textWidth = font.getWidth("");
+            for (String singleLine : linesOfText) {
+                if (font.getWidth(singleLine) > textWidth)
+                    textWidth = font.getWidth(singleLine);
             }
-            int textWidth = font.getWidth(longest);
 
             FontMeshBuilder meshBuilder = new FontMeshBuilder(underlineMaterial);
 
             Map<Material, Mesh> meshMap = entityMeshCache.get(entity);
             if (meshMap == null) {
                 meshMap = meshBuilder
-                        .createTextMesh(font, text, textWidth, HorizontalAlign.CENTER, baseColor,
+                        .createTextMesh(font, Arrays.asList(linesOfText), textWidth, HorizontalAlign.CENTER, baseColor,
```

Comment:
Ok, looking at the code I can't see a reason why the createTextMesh() method can't accept a string array instead. Can you take a look? It will cascade in a few other changes but they might still be safe to do.